libs/SalesforceSDK/src/com/salesforce/androidsdk/auth/OAuth2.java

```diff
@@ -649,7 +652,7 @@ private void populateFromJSON(JSONObject parsedResponse) {
                 userId = parsedResponse.getString(USER_ID);
                 orgId = parsedResponse.getString(ORG_ID);
                 nickname = parsedResponse.getString(NICKNAME);
-                final JSONObject urls = parsedResponse.getJSONObject(URLS);
+                final JSONObject urls = parsedResponse.optJSONObject(URLS);
```

Comment:
Old code - if it can get null, we need to to optJSONObject not getJSONObject (which throws).
Because of the catch below, the code will continue with all the fields below the throwing line not set :-(